site/tests/integration/sign-up-flow.ts

```diff
@@ -0,0 +1,114 @@
+import { expect, test } from "playwright-test-coverage";
+
+import { readValueFromRecentDummyEmail } from "../shared/dummy-emails";
+import { resetDb } from "../shared/fixtures";
+import { login, openMobileNav } from "../shared/nav";
+
+test("sign up flow works", async ({ browserName, isMobile, page }) => {
+  await resetDb();
+  await page.goto("/");
+
+  if (isMobile) {
+    await openMobileNav(page);
+  }
+
+  await page
+    .locator(isMobile ? "a:has-text('Sign Up')" : "header >> text=Sign Up")
+    .click();
+
+  await expect(page).toHaveURL("/signup");
+
+  await expect(
+    page.locator("text=Create your Block Protocol account"),
+  ).toBeVisible();
+
+  await expect(
+    page.locator("text=Publish blocks to the block hub"),
+  ).toBeVisible();
+
+  await expect(
+    page.locator("text=Take part in a growing, open source community"),
+  ).toBeVisible();
+
+  await expect(page.locator("text=Claim your favorite username")).toBeVisible();
+
+  const emailInput = page.locator('[placeholder="claude\\@shannon\\.com"]');
+  await emailInput.fill("alice@example.com");
+  await emailInput.press("Enter");
+
+  await expect(
+    page.locator("text=An existing user is associated with the email address"),
+  ).toBeVisible();
+
+  await emailInput.fill("alice@example.org");
+
+  const continueButton = page.locator("text=Continue");
+  await continueButton.click();
+  await expect(page).toHaveURL("/signup");
+
+  await expect(page.locator("text=Check your inbox")).toBeVisible();
+  const verificationCodeInput = page.locator(
+    '[placeholder="your-verification-code"]',
+  );
+  await verificationCodeInput.fill(
+    await readValueFromRecentDummyEmail("Email verification code: "),
+  );
+  await verificationCodeInput.press("Enter");
+
+  await expect(page.locator("text=alice@example.org")).toBeVisible();
+  await expect(page.locator("text=Add your account details")).toBeVisible();
+  await expect(page).toHaveURL("/signup");
+
+  const usernameInput = page.locator('[placeholder="claudeshannon"]');
+  await usernameInput.fill("alice");
+  await page.locator('[placeholder="claudeshannon"]').press("Tab");
+
+  await expect(
+    page.locator("text=This user has already been taken"),
+  ).toBeVisible();
+
+  await usernameInput.fill("alice2");
+
+  const preferredNameInput = page.locator('[placeholder="Claude"]');
+  await preferredNameInput.click();
+
+  await expect(
+    page.locator('text=This user has already been taken"'),
+  ).not.toBeVisible();
+
+  await preferredNameInput.fill("Alice");
+
+  // @todo Remove when Safari bug is fixed
+  // (main page scrolls down when modal is closed, navbar disappears)
+  test.skip(
```

Comment:
I assume test.skip should be used if we want to skip the entire test as opposed to a section. I think it's better to use a conditional to handle the remaining parts of the test

```
if (browserName !== "webkit")
{
  ...
}
```

Reply:
This can be addressed in a follow up

Reply:
WDYT of using `test.fixme` instead?

Reply:
Yeah we should definitely use that